src/NHibernate/Id/SelectGenerator.cs

```diff
@@ -95,31 +71,65 @@ protected internal override SqlString SelectSQL
 
 			protected internal override SqlType[] ParametersTypes
 			{
-				get { return uniqueKeyType.SqlTypes(factory); }
+				get
+				{
+					return uniqueKeyTypes.SelectMany(t => t.SqlTypes(factory)).ToArray();
+				}
 			}
 
 			public override IdentifierGeneratingInsert PrepareIdentifierGeneratingInsert()
 			{
 				return new IdentifierGeneratingInsert(factory);
 			}
 
+			// Since 5.2
+			[Obsolete("Use or override BindParameters(ISessionImplementor session, DbCommand ps, IBinder binder) instead.")]
 			protected internal override void BindParameters(ISessionImplementor session, DbCommand ps, object entity)
 			{
-				object uniqueKeyValue = ((IEntityPersister) persister).GetPropertyValue(entity, uniqueKeyPropertyName);
-				uniqueKeyType.NullSafeSet(ps, uniqueKeyValue, 0, session);
+				var entityPersister = (IEntityPersister) persister;
+				var uniqueKeyPropertyNames = uniqueKeySuppliedPropertyNames ??
+					PostInsertIdentityPersisterExtension.DetermineNameOfPropertiesToUse(entityPersister);
+				for (var i = 0; i < uniqueKeyPropertyNames.Length; i++)
+				{
+					var uniqueKeyValue = entityPersister.GetPropertyValue(entity, uniqueKeyPropertyNames[i]);
+					uniqueKeyTypes[i].NullSafeSet(ps, uniqueKeyValue, i, session);
+				}
+			}
+
+			protected internal override void BindParameters(ISessionImplementor session, DbCommand ps, IBinder binder)
+			{
+				// 6.0 TODO: remove the "if" block and do the other TODO of this method
+				if (persister is IEntityPersister)
+				{
+#pragma warning disable 618
+					BindParameters(session, ps, binder.Entity);
```

Comment:
This is required to avoid a possible breaking change in the unlikely case someone uses a class overriding the obsoleted method. This causes `BindSelectByUniqueKey` of entity persisters to be currently not used.

Still more unlikely, this is maybe not enough: what if someone had implemented deriving classes correctly supporting idbag collections? But I do not see how we could handle that.